### PR TITLE
refactor to a client, common library, and server

### PR DESCRIPTION
**Pull request by rphillips**
Not sure if we want to do this... but it may cleanup the file layout a bit and enforce public interfaces.

```
.
├── Cargo.lock
├── Cargo.toml
├── conmon-rs
│  ├── client
│  │  ├── Cargo.toml
│  │  └── src
│  │     └── main.rs
│  ├── common
│  │  ├── build.rs
│  │  ├── Cargo.toml
│  │  ├── proto
│  │  │  └── conmon.capnp
│  │  └── src
│  │     └── lib.rs
│  └── server
│     ├── Cargo.toml
│     └── src
│        ├── config.rs
│        ├── init.rs
│        └── main.rs
├── conmon-rs-tests
│  ├── Cargo.toml
│  └── tests
│     └── lib.rs
├── LICENSE
├── Makefile
└── README.md
```

### CODE REVIEW

**Comment by haircommander** [comment]
yeah I think this is a good idea :+1:

**Comment by rphillips** [comment]
@saschagrunert @haircommander ready for review

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -138,11 +138,13 @@ jobs:
       - name: Select Toolchain
         uses: actions-rs/toolchain@v1
         with:
-          toolchain: stable
+          toolchain: nightly
```
Comment (haircommander) [inline .github/workflows/ci.yml:141]: what's the reason for this change?

Comment (rphillips) [reply .github/workflows/ci.yml:141]: we build everything with nightly... this, I think, was an oversight

Comment (haircommander) [reply .github/workflows/ci.yml:141]: ah I see

**Comment by haircommander** [comment]
LGTM, @saschagrunert for the final merge?

**Review by saschagrunert — APPROVED** [review]
